### PR TITLE
fix(core/changelogs): pass though known project info

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Currently every generic changelog retrival code tries to detect the platform.
With this change we pass the already known platform info to all lower levels, so we can simply skip platoform detection.

This also fixes wrong platform detection by examining hostRules.
If there is a rule for `github` without `hostType`  renovate would use the `gitlab` http layer instead of the `github` http layer.

Additionally i've added the hostType to the http error logs, so we can hopefully see mismatches later.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/pr/changelog/release-notes.ts [hunk 1]
```diff
@@ -2,64 +2,58 @@ import * as URL from 'url';
...
     } else {
-      logger.info({ repository, err }, 'getReleaseList error');
+      logger.debug(
+        { repository, type, apiBaseUrl, err },
+        'getReleaseList error'
+      );
     }
-    return [];
   }
+  return [];
```
Comment (rarkins) [inline lib/workers/pr/changelog/release-notes.ts:44]: Should we log here that no supported type was found?

File: lib/workers/pr/changelog/release-notes.ts [hunk 2]
```diff
@@ -172,84 +164,67 @@ function isUrl(url: string): boolean {
...
     } else {
-      logger.debug({ err, repository }, 'Error getting changelog md');
+      logger.debug(
+        { err, repository, type, apiBaseUrl },
+        'Error getting changelog md'
+      );
     }
-    return null;
   }
+  return null;
```
Comment (rarkins) [inline lib/workers/pr/changelog/release-notes.ts:199]: log

**Review by rarkins — COMMENTED** [review]
How widely could this be tested so far? e.g. github.com, self-hosted gitlab, gitlab.com, etc?

File: lib/workers/pr/changelog/__snapshots__/release-notes.spec.ts.snap [hunk 3]
```diff
@@ -189,16 +191,7 @@ Array [
 ]
 `;
 
-exports[`workers/pr/changelog/release-notes getReleaseNotes() gets release notes with body 5`] = `
```
Comment (viceice) [inline lib/workers/pr/changelog/__snapshots__/release-notes.spec.ts.snap:192]: 🤔 need to check this

File: lib/workers/pr/changelog/release-notes.ts (hunk 2 above)
Comment (viceice) [reply lib/workers/pr/changelog/release-notes.ts:205]: done

File: lib/workers/pr/changelog/release-notes.ts (hunk 1 above)
Comment (viceice) [reply lib/workers/pr/changelog/release-notes.ts:47]: done

**Comment by viceice** [comment]
Should also work for github enterprise

File: lib/workers/pr/changelog/__snapshots__/release-notes.spec.ts.snap [hunk 4]
```diff
@@ -267,9 +269,9 @@ Array [
 ]
 `;
 
-exports[`workers/pr/changelog/release-notes getReleaseNotes() gets release notes with body from gitlab repo  1`] = `null`;
+exports[`workers/pr/changelog/release-notes getReleaseNotes() gets release notes with body from gitlab repo "" 1`] = `null`;
```
Comment (viceice) [inline lib/workers/pr/changelog/__snapshots__/release-notes.spec.ts.snap:272]: This looks still like a wrong test, or some missing stuff

File: lib/workers/pr/changelog/release-notes.ts [hunk 5]
```diff
@@ -172,84 +167,70 @@ function isUrl(url: string): boolean {
...
-  sourceDirectory?: string
+  project: ChangeLogProject,
+  version: string
 ): Promise<ChangeLogNotes | null> {
+  const { baseUrl, repository } = project;
   logger.trace(`getReleaseNotesMd(${repository}, ${version})`);
   const skippedRepos = ['facebook/react-native'];
   // istanbul ignore if
-  if (skippedRepos.includes(repository)) {
+  if (skippedRepos.includes(project.repository)) {
```
Comment (viceice) [inline lib/workers/pr/changelog/release-notes.ts:230]:
```suggestion
  if (skippedRepos.includes(repository)) {
```

**Review by rarkins — DISMISSED** [review]

File: lib/workers/pr/changelog/gitlab/index.ts [hunk 6]
```diff
@@ -107,14 +108,7 @@ export async function getReleaseList(
   )}projects/${repoId}/releases`;
 
   // TODO: use type from PR #11226
```
Comment (HonkingGoose) [inline lib/workers/pr/changelog/gitlab/index.ts:110]: Is this TODO still relevant now that you've removed/edited the previous code?

Comment (viceice) [reply lib/workers/pr/changelog/gitlab/index.ts:110]:
```suggestion
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**